go/services/reports/pkg/incident_correlation/incident_merger.go

```diff
@@ -0,0 +1,253 @@
+		client:                  params.Client,
+		logger:                  params.Logger,
+		batchPoller:             params.BatchPoller,
+		scheduledJobsRepository: params.ScheduledJobsRepository,
+	}
+}
+
+func (m *OpenAiIncidentMerger) getIncidentMergeCriteria(incident repositories.Incident) *IncidentMergeCriteria {
+	return &IncidentMergeCriteria{
+		Id:             incident.GetId(),
+		Title:          incident.GetTitle(),
+		Summary:        incident.GetSummary(),
+		Recommendation: incident.GetRecommendation(),
+		Urgency:        incident.GetUrgency(),
+		Category:       incident.GetCategory(),
+	}
+}
+
+func (m *OpenAiIncidentMerger) encodeMergeCriteria(incidents []*IncidentMergeCriteria) ([]string, error) {
+
+	var encodedSummaries = make([]string, 0, len(incidents))
+
+	for _, summary := range incidents {
+		encodedSummary, err := json.Marshal(summary)
+		if err != nil {
+			m.logger.Error("Failed to encode incident summary", zap.Error(err), zap.Any("incident", summary))
+			return nil, err
+		}
+
+		encodedSummaries = append(encodedSummaries, string(encodedSummary))
+	}
+
+	return encodedSummaries, nil
+}
+
+func (m *OpenAiIncidentMerger) createSummaryRequestMessage(content []string) []*openai.Message {
+
+	return []*openai.Message{
+		{
+			Role:    "system",
+			Content: "You are a Kubernetes cluster system administrator.",
+		},
+		{
+			Role: "user",
+			Content: fmt.Sprintf(`
+    Given a list of incident reports, each with a summary, title, and recommendation for resolution, identify and group incidents 
+    likely related to the same underlying event. For each incident:
+    Title and Summary Similarity: Analyze the titles and summaries to identify incidents describing similar issues. Incidents with overlapping language, 
+    key terms, or repeated phrases should be flagged as potentially related.
+    Recommendation Consistency: Compare recommended resolutions. Incidents with highly similar or identical recommendations are 
+    likely related, as they suggest a common resolution approach.
+    Semantic Understanding: Use natural language processing to understand the contextual and semantic meaning of each summary and title. 
+    Incidents that describe similar issues and solutions in different wording should still be grouped together based on meaning.
+    Event Grouping: For each group of related incidents, assign a unique event identifier and 
+    list all included incidents with their title, summary, and recommendation, noting the factors contributing to their grouping.
+        %s`, content),
+		},
+	}
+}
+
+type IncidentMergeGroup struct {
+	Title          string           `json:"title"`
+	Summary        string           `json:"summary"`
+	Recommendation string           `json:"recommendation"`
+	Category       string           `json:"category"`
+	Urgency        insights.Urgency `json:"urgency"`
+	IncidentIds    []string         `json:"incidentIds"`
+}
+
+type incidentMergerResponseDto struct {
```

Comment:
Is this type lowercase intentionally?

Reply:
Yes, this is intentional, it should be visible only in the scope of this package